xray/services/ignorerule.go

```diff
@@ -0,0 +1,140 @@
+// IgnoreRuleService defines the http client and Xray details
+type IgnoreRuleService struct {
+	client      *jfroghttpclient.JfrogHttpClient
+	XrayDetails auth.ServiceDetails
+}
+
+// NewIgnoreRuleService creates a new Xray Ignore Rule Service
+func NewIgnoreRuleService(client *jfroghttpclient.JfrogHttpClient) *IgnoreRuleService {
+	return &IgnoreRuleService{client: client}
+}
+
+// GetXrayDetails returns the Xray details
+func (xirs *IgnoreRuleService) GetXrayDetails() auth.ServiceDetails {
+	return xirs.XrayDetails
+}
+
+// GetJfrogHttpClient returns the http client
+func (xirs *IgnoreRuleService) GetJfrogHttpClient() *jfroghttpclient.JfrogHttpClient {
+	return xirs.client
+}
+
+// The getIgnoreRuleURL does not end with a slash
+// So, calling functions will need to add it
+func (xirs *IgnoreRuleService) getIgnoreRuleURL() string {
+	return clientutils.AddTrailingSlashIfNeeded(xirs.XrayDetails.GetUrl()) + ignoreRuleAPIURL
+}
+
+// Delete will delete an ignore rule by id
+func (xirs *IgnoreRuleService) Delete(ignoreRuleId string) error {
+	httpClientsDetails := xirs.XrayDetails.CreateHttpClientDetails()
+	artUtils.SetContentType("application/json", &httpClientsDetails.Headers)
+
+	log.Info("Deleting ignore rule...")
+	resp, body, err := xirs.client.SendDelete(xirs.getIgnoreRuleURL()+"/"+ignoreRuleId, nil, &httpClientsDetails)
+	if err != nil {
+		return err
+	}
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusNoContent); err != nil {
+		return err
+	}
+	log.Debug("Xray response status:", resp.Status)
+	log.Info("Done deleting ignore rule.")
+	return nil
+}
+
+// Create will create a new Xray ignore rule
+// The function creates the ignore rule and returns its id which is recieved after post
+func (xirs *IgnoreRuleService) Create(params utils.IgnoreRuleParams) (ignoreRuleId string, err error) {
+	ignoreRuleBody := utils.CreateIgnoreRuleBody(params)
+	content, err := json.Marshal(ignoreRuleBody)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+
+	httpClientsDetails := xirs.XrayDetails.CreateHttpClientDetails()
+	artUtils.SetContentType("application/json", &httpClientsDetails.Headers)
+	var url = xirs.getIgnoreRuleURL()
+
+	log.Info("Create new ignore rule...")
+	resp, body, err := xirs.client.SendPost(url, content, &httpClientsDetails)
+	if err != nil {
+		return "", err
+	}
+
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusCreated); err != nil {
+		return "", err
+	}
+	log.Debug("Xray response status:", resp.Status)
+
+	ignoreRuleId, err = getIgnoreRuleIdFromBody(body)
+	if err != nil {
+		return "", err
+	}
+
+	log.Info("Done creating ignore rule.")
+	log.Debug("Ignore rule id is: ", ignoreRuleId)
+
+	return ignoreRuleId, nil
+}
+
+func getIgnoreRuleIdFromBody(body []byte) (string, error) {
+	str := string(body)
+
+	re := regexp.MustCompile(`id:\s*([a-f0-9-]+)`)
+	match := re.FindStringSubmatch(str)
+
+	if len(match) <= 1 {
+		err := errors.New("couldn't find id for ignore rule")
```

Comment:
I recommend adding a more specific error message tp the error. Perhaps include the string that failed to be matched to the error.
In addition, this error should be wrapped with CheckError. See other examples in the code for this.